vizro-core/src/vizro/models/_components/form/range_slider.py

```diff
@@ -54,32 +54,55 @@ def build(self):
         ]
-        input = [
+        inputs = [
             Input(f"{self.id}_start_value", "value"),
             Input(f"{self.id}_end_value", "value"),
             Input(self.id, "value"),
             State(f"temp-store-range_slider-{self.id}", "data"),
+            State(f"{self.id}_callback_data", "data"),
         ]
 
-        @callback(output=output, inputs=input)
-        def update_slider_values(start, end, slider, input_store):
-            trigger_id = callback_context.triggered_id
-            if trigger_id == f"{self.id}_start_value" or trigger_id == f"{self.id}_end_value":
-                start_text_value, end_text_value = start, end
-            elif trigger_id == self.id:
-                start_text_value, end_text_value = slider
-            else:
-                start_text_value, end_text_value = input_store if input_store is not None else value
-
-            start_value = min(start_text_value, end_text_value)
-            end_value = max(start_text_value, end_text_value)
-            start_value = max(self.min, start_value)
-            end_value = min(self.max, end_value)
-            slider_value = [start_value, end_value]
-            return start_value, end_value, slider_value, (start_value, end_value)
+        clientside_callback(
+            """
+            function update_slider_values(start, end, slider, input_store, self_data) {
+                var end_text_value, end_value, slider_value, start_text_value, start_value, trigger_id;
+
+                trigger_id = dash_clientside.callback_context.triggered
+                if (trigger_id.length != 0) {
+                    trigger_id = dash_clientside.callback_context.triggered[0]['prop_id'].split('.')[0];
+                }
+                if (trigger_id === `${self_data["id"]}_start_value` || trigger_id === `${self_data["id"]}_end_value`) {
+                    [start_text_value, end_text_value] = [start, end];
+                } else if (trigger_id === self_data["id"]) {
+                    [start_text_value, end_text_value] = [slider[0], slider[1]];
+                } else {
+                  [start_text_value, end_text_value] = input_store !== null ? input_store : [slider[0], slider[1]];
+                }
+
+                start_value = Math.min(start_text_value, end_text_value);
+                end_value = Math.max(start_text_value, end_text_value);
+                start_value = Math.max(self_data["min"], start_value);
+                end_value = Math.min(self_data["max"], end_value);
+                slider_value = [start_value, end_value];
+
+                return [start_value, end_value, slider_value, [start_value, end_value]];
+            }
+            """,
+            output=output,
+            inputs=inputs,
+        )
 
         return html.Div(
             [
+                dcc.Store(
+                    f"{self.id}_callback_data",
+                    storage_type="local",
```

Comment:
what's the implication when `storage_type="local"`? 

I see

> storage_type (a value equal to: 'local', 'session' or 'memory'; default 'memory'): The type of the web storage. memory: only kept in memory, reset on page refresh. local: window.localStorage, data is kept after the browser quit. session: window.sessionStorage, data is cleared once the browser quit.

from https://dash.plotly.com/dash-core-components/store

Would it keep stale data in the Store?

Reply:
As it's explained in the link you sent, data will be preserved in the store if users refresh the currently opened page in the app or even if they close and reopen a browser. But, data in the `f"{self.id}_callback_data"` can't be changed (once it's returned from the server) so it doesn't make sense to keep in the `window.localStorage`. 


I removed the `storage_type` here and in the `Slider` as well. Thanks for the heads up 🚀.